src/briefcase/commands/base.py

```diff
@@ -117,24 +117,70 @@ class BaseCommand(ABC):
     def __init__(
         self,
         base_path,
-        home_path=Path.home(),
-        data_path=PlatformDirs(appname="briefcase", appauthor="BeeWare").user_data_path,
+        home_path=None,
+        data_path=None,
         apps=None,
         input_enabled=True,
     ):
+        # Some details about the host machine
+        self.host_arch = platform.machine()
+        self.host_os = platform.system()
+
         self.base_path = base_path
-        self.home_path = home_path
-        self.data_path = data_path
+
+        # If a home path is provided during construction, use it. This usually
+        # indicates we're under test conditions.
+        if home_path is None:
+            home_path = Path.home()
+        self.home_path = Path(home_path)
+
+        # If a data path is provided during construction, use it. This usually
+        # indicates we're under test conditions. If there's no data path
+        # provided, look for a BRIEFCASE_HOME environment variable. If that
+        # isn't defined, use a platform-specific default data path.
+        if data_path is None:
+            try:
+                data_path = Path(os.environ["BRIEFCASE_HOME"])
+                if not data_path.exists():
```

Comment:
Unfortunately, `Path("")` returns `PosixPath('.')`....so this lets an empty `BRIEFCASE_HOME` sneak by. I wouldn't consider this especially important if I hadn't tried `export <ENV VAR>=""` too many times in the past instead of `unset <ENV VAR>`.

Reply:
Good catch - I've added an explicit catch of this case.